packages/@lwc/integration-not-karma/helpers/lwc.mjs

```diff
@@ -0,0 +1,97 @@
+import path from 'node:path';
+import { rollup } from 'rollup';
+import lwcRollupPlugin from '@lwc/rollup-plugin';
+
+import {
+    DISABLE_SYNTHETIC_SHADOW_SUPPORT_IN_COMPILER,
+    API_VERSION,
+    DISABLE_STATIC_CONTENT_OPTIMIZATION,
+} from './options.mjs';
+
+// Cache reused between each compilation to speed up the compilation time.
+let cache;
+
+export default async (ctx) => {
+    const input = ctx.path.slice(1); // strip leading / from URL path to get relative file path
+    const suiteDir = path.dirname(input);
+
+    // TODO [#3370]: remove experimental template expression flag
+    const experimentalComplexExpressions = suiteDir.includes('template-expressions');
+
+    const createRollupPlugin = (options) => {
+        return lwcRollupPlugin({
+            // Sourcemaps don't work with Istanbul coverage
+            sourcemap: !process.env.COVERAGE,
+            experimentalDynamicComponent: {
+                loader: 'test-utils',
+                strict: true,
+            },
+            enableDynamicComponents: true,
+            enableLwcOn: true,
+            experimentalComplexExpressions,
+            enableStaticContentOptimization: !DISABLE_STATIC_CONTENT_OPTIMIZATION,
+            disableSyntheticShadowSupport: DISABLE_SYNTHETIC_SHADOW_SUPPORT_IN_COMPILER,
+            apiVersion: API_VERSION,
+            modules: [
+                {
+                    // Assume `ctx.path` is a component file, e.g. modules/x/foo/foo.js
+                    dir: path.resolve(input, '../../..'),
+                },
+            ],
+            ...options,
+        });
+    };
+
+    const defaultRollupPlugin = createRollupPlugin();
```

Comment:
Yeah this is a bit weird. It's okay for now, but it seems like something that the rollup plugin should be able to handle for us, without creating three different versions of the plugin.